pkg/skaffold/deploy/deploy.go

```diff
@@ -38,7 +37,7 @@ type Deployer interface {
 
 	// Deploy should ensure that the build results are deployed to the Kubernetes
 	// cluster.
-	Deploy(context.Context, io.Writer, []build.Artifact) ([]Artifact, error)
+	Deploy(context.Context, io.Writer, []build.Artifact, []Labeller) error
```

Comment:
instead of passing the labellers as a parameter to the `Deploy()` call, could we just embed the labellers themselves into the deployer object when we instantiate the deployer in the runner? imo the simpler the method signature the better, especially for something that might be exposed externally (for plugins) in the future

Reply:
I tried to go that way first but I felt like labelling is really part of the Deployer's contract so should be visible here.

Reply:
wdyt?

Reply:
mmm ok I see your reasoning. sgtm